src/app/data-client.test.ts

```diff
@@ -0,0 +1,263 @@
+      expect(methodSpy).toHaveBeenCalledWith(
+        req,
+        expect.anything(),
+        expect.anything()
+      );
+    });
+  });
+
+  describe('binaryDataById tests', () => {
+    let methodSpy: SpyInstance;
+    beforeEach(() => {
+      methodSpy = vi
+        .spyOn(DataServiceClient.prototype, 'binaryDataByIDs')
+        // @ts-expect-error compiler is matching incorrect function signature
+        .mockImplementation((_req: BinaryDataByIDsRequest, _md, cb) => {
+          cb(null, binDataResponse);
+        });
+    });
+
+    const binaryId1 = new BinaryID();
+    binaryId1.setFileId('testFileId1');
+    binaryId1.setOrganizationId('testOrgId');
+    binaryId1.setLocationId('testLocationId');
+    const binaryId2 = new BinaryID();
+    binaryId2.setFileId('testFileId1');
+    binaryId2.setOrganizationId('testOrgId');
+    binaryId2.setLocationId('testLocationId');
+
+    test('get binary data by ids', async () => {
+      const promise = await subject().binaryDataByIds([
+        binaryId1.toObject(),
+        binaryId2.toObject(),
+      ]);
+      expect(promise.length).toEqual(2);
+      expect(promise[0]?.binary).toEqual(bin1);
+      expect(promise[1]?.binary).toEqual(bin2);
+    });
+
+    test('get binary data by id', async () => {
+      const req = new BinaryDataByIDsRequest();
+      req.setBinaryIdsList([binaryId1]);
+      req.setIncludeBinary(true);
+
+      await subject().binaryDataByIds([binaryId1.toObject()]);
+      expect(methodSpy).toHaveBeenCalledWith(
+        req,
+        expect.anything(),
+        expect.anything()
+      );
+    });
+  });
+
+  describe('createFilter tests', () => {
+    test('create empty filter', () => {
+      const testFilter = subject().createFilter({});
+      expect(testFilter).toEqual(new Filter());
+    });
+
+    test('create filter', () => {
+      const opts = { componentName: 'camera' };
+      const testFilter = subject().createFilter(opts);
+
+      const actualFilter = new Filter();
+      actualFilter.setComponentName('camera');
+
+      expect(testFilter).toEqual(actualFilter);
+    });
+
+    test('create filter with all options', () => {
+      const componentName = 'testComponentName';
+      const componentType = 'testComponentType';
+      const method = 'testMethod';
+      const robotName = 'testRobotName';
+      const robotId = 'testRobotId';
+      const partName = 'testPartName';
+      const partId = 'testPartId';
+      const locationsIdsList = ['testLocationId1', 'testLocationId2'];
+      const organizationIdsList = ['testOrgId1', 'testOrgId2'];
+      const mimeTypeList = ['testMimeType1', 'testMimeType2'];
+      const bboxLabelsList = ['testBboxLabel1', 'testBboxLabel2'];
+      const startTime = new Date(1, 1, 1, 1, 1, 1);
+      const endTime = new Date(2, 2, 2, 2, 2, 2);
+      const interval = new CaptureInterval();
+      interval.setStart(Timestamp.fromDate(startTime));
+      interval.setEnd(Timestamp.fromDate(endTime));
```

Comment:
let's make it so the `createFilter` function accepts `CaptureInterval.asObject`, so it can be defined as a plain object, e.g.

```suggestion
      const interval = { start: startTime, end: endTime };
```

Reply:
ugh, ignore this! this is already the case - i mixed up the test with the expections